EyeSeeTea/user-extended-app
# feature: export migration

### :pushpin: References

-   **Issue:** https://app.clickup.com/t/8694mneh7
- 8694mneh7

### :memo: Implementation
⚠️ This PR only concern **export**, the **import** will be done in a separate PR (it'll be to big)
- new `ImportExport.tsx` component (old `ImportExport.component.js` is deleted)
- new `ExportUsersUseCase` that migrate related legacy function in `userHelpers.js`
- tests for use case
- custom hook `useExportUsers`
- When orgUnitsField (ex: code) doesn't exist it fallback to `name` field (I encounter a org unit without a `code` and it export `null` so I think it could be a workaround to use name)

### :video_camera: Screenshots/Screen capture

https://github.com/EyeSeeTea/user-extended-app/assets/6376073/82c67b2b-a292-4e09-be8e-3747230b3a48


### :fire: Testing
- Try to export in CSV and JSON
- Filter with name for example and try to export again, you should only have the filtered users
- Import should still work
- `yarn test` should still pass
